src/index.adoc

```diff
@@ -4,14 +4,542 @@ include::header.adoc[]
+of people don't live this elementary rule, and they waste a lot of their time.
+
+=== link:https://www.youtube.com/watch?v=o9pEzgHorH0[Jack Diederich: Stop
+Writing Classes^] Despite its name, the talk doesn't imply that OOP is bad.
+Rather than that, it promotes writing short, readable code, avoiding premature
+generalizations, working with pure values, and avoiding complex OOP structures.
+It's funny and refreshing talk.
+
+=== link:https://www.youtube.com/watch?v=QM1iUe6IofM[Brian Will: Why OOP is Bad^]
+Brian gives a compelling argument that state encapsulation doesn't work: The
+real systems are dominated by cross-cutting concerns. When faced with the
+reality of real word programs, promises of encapsulation tend to be broken, and
+object graphs quickly become tangled webs.
+
+Apart from this, Brian does a very systematic breakdown of what Java has brought to engineering
+world, what are some good parts of it and why it thrives even though OOP is
+bad.
+
+=== link:https://www.youtube.com/watch?v=JjDsP5n2kSM&t=2169s[Jonathan Blow: How
+to program independent games^] You don't succeed by using fancy data structure or
+writing an algorithm that's a bit faster but buggy and fragile. Independent game
+developer needs to be "extremely effective at getting things done".
+
+=== Rich Hickey on immutable values versus references
+
+- link:https://www.infoq.com/presentations/Value-Values[The Value of Values^]
+- link:https://www.infoq.com/presentations/Are-We-There-Yet-Rich-Hickey[Are we
+  there yet^]
+
+This is Rich (again!) promoting immutable values and immutable data structures
+over references that change over time in a mutable manner. Also, Rich talks
+about the Epochal time model, where the whole (!) state of the program is
+represented by a single mutable reference that takes a sequence of immutable
+values, modeling a time-evolution of a state. This model is used with Redux
+framework.
+
+=== link:https://www.youtube.com/watch?v=FihU5JxmnBg[Stuart Halloway: Debugging
+with the Scientific Method^]
+
+Be less chaotic and more organized when debugging. State explicitly what's your
+hypothesis, test it, think. Write things down. Test one thing at a time. A good
+process is more important than tools; scientific approach with console.log is
+more powerful than ad-hoc trying random stuff with a strong debugger.
 
 == Tools
-include::tools/index.adoc[]
+
+=== Operating System
+Use either Mac or Linux for development. Don't use Windows, don't even think
+about it! There is always something working differently on Windows than on the
+other platforms. This costs us huge amounts of time debugging problems instead
+of developing valuable features.
+
+If you are new to Linux, install link:http://www.ubuntu.com[Ubuntu^]. Make sure to read
+link:https://help.ubuntu.com/stable/ubuntu-help[the introduction to Ubuntu Desktop^], especially
+link:https://help.ubuntu.com/stable/ubuntu-help/addremove.html[how the software installation work in Linux^].
```

Comment:
I would even go as far as recommend manjaro over Ubuntu for newcomers. Better installation, better package manager with more packages out of the box. The default DEs KDE/XFCE >>> gnome for anyone new to linux. (gnome a very unintuitive UX for anyone coming from windows)

Reply:
Well, I would recommend against Manjaro. It's a rolling distro and don't forget that can be often tricky even for power users. I am, well, not a Linux guru but a long time user that tried many distros.

Manjaro is maybe for example a good gaming distro but in my year-long experience it got broken a few times and I had to fix it manually.
There is AUR which is good for users like me but stuff there is often broken too and one has to edit install scripts. Moreover the packages there work against Arch packages and Manjaro is 1-2 weeks late compared to Arch - and that breaks stuff too.

Yes, many of us are against Ubuntu because of various reasons but hey, it's still a good, "simple" distro for beginners together with Mint.
They can easily switch to their own choice when they would be capable of doing that.

Reply:
I had been using ArchLinux in past, and there was some amount of time spent due to things being broken by updates. Since I moved to Ubuntu, maintenance costs / time spent was heavily reduced. I am not heavily chasing performance anymore, though.

If I was to switch to something else, which would be more complex than Ubuntu, then I would consider Qubes OS. It would give me more value (increased security) for the extra maintenance cost. It might even reduce costs in long-term, as it allows to separate project specific tools, and easily drop them completely (delete qube / virtual machine) without a need to reinstall OS to be sure there aren't any leftovers.